nasa-jpl/explorer-1
# Adding an internal theme

### Checklist

  - PR titles are used to generate the change log in [releases](../releases); good ones make that easier to scan.
  - PRs will be broadly categorized in the change log, but for even easier scanning, consider prefixing with a component name or other useful categorization, e.g., "BaseButton: fix layout bug", or "Storybook: Update dependencies".
  - These tags are used to do the aforementioned broad categorization in the change log and determine what the next release's version number should be.
  - [Release Drafter](https://github.com/marketplace/actions/release-drafter) will attempt to do the category labeling for you! Please double-check its work.

## Description
This is a start to "combining the storybooks." It really only covers adding the internal styles. Some implementation notes:

- I went with the [approach discussed in #9](https://github.com/nasa-jpl/explorer-1/issues/9#issuecomment-1067194814). Users who want to use the internal theme will apply a body class of `ThemeInternal` to their site.
- Unfortunately, we did not have a `.BaseLink` class in place on our BaseLink component template, so users that want to start using the internal theme will need to update a lot of their component templates. On the other hand, we may not have that many users, so perhaps not a big worry.
- Currently, the internal theme styles are separated by component (added to existing component scss files). I went back and forth between this and the alternate approach of having all internal styles in a dedicated SCSS file. At the time, it seemed more sensible to have component SCSS grouped together for comparing rules, but it does have the unfortunate effect of sometimes needing to add a SCSS file for a component just for the internal theme (ex: `_BlockAccordion.scss`). It will also make it more difficult to create a dedicated `internal.min.css` file as discussed. I think I'm going to change the approach and move all internal theme styles to a single SCSS file, but I'm definitely interested in other devs thoughts on this before I make that change.
- The documentation for how to use the internal theme is pretty simple. See the [Foundation: Themes](http://localhost:6006/?path=/docs/foundation-themes--internal) page in the storybook.
- When we add internal and external components, I'm not planning on creating separate categories. I prefer @Scotchester's reccommendation to specify `HeaderInternal` and `HeaderExternal`. Then in storybook we can structure the documentation like so:

```
Components
└── Header
    ├── External
    └── Internal
```

## Instructions to test

1. `npm run storybook`
2. View a component, or even a component overview page. Use the theme switcher to use the internal theme. Note that any stories using an iframe to display a component on a docs page will ignore the theme switcher (usually anything that requires additional javascript). I'd really like to fix that issue in storybook, as it also leads to [this problem](https://github.com/nasa-jpl/explorer-1/issues/145). Hoping to get that prioritized soon.
3. See the documentation on how to implement the internal theme on your project on the [Foundation: Themes](http://localhost:6006/?path=/docs/foundation-themes--internal) page.

## Tested in the following environments/browsers:



### Operating System


### Browser
